### ISSUE TITLE
Wrong section cropped?

### ISSUE THREAD

**Issue by youradds** [Foliotek/Croppie#125]
Hi,

I've got a bit of a weird issue here! 

Below is my JS:

```
var $uploadCrop;
$(document).ready( function() {

			$uploadCrop = $('#uploader-avatar').croppie({
			    // enableExif: true,
			    viewport: {
			        width: 200,
			        height: 200,
			        type: 'square'
			    },
			    boundary: {
			        width: 300,
			        height: 300
			    }
			});
			$uploadCrop.bind();


	$('#upload').on('click', function () {
		$('#view-avatar').hide();
		if (typeof $uploadCrop != "object") {

		}
	});

	$('#upload').on('change', function () { readFile(this); });

	$('.upload-result-avatar').on('click', function (ev) {
		$uploadCrop.croppie(
		'result', {
			type: 'canvas',
			size: 'viewport',
			format: 'jpeg'
		}).then(function (resp) {
			console.log(resp);
			$('#view-avatar').show();
			$('#currentAvatar').show().attr('src',resp);

		});
	});

});

function readFile(input) {

		if (input.files && input.files[0]) {

        var reader = new FileReader();

        reader.onload = function (e) {
        	$uploadCrop.croppie('bind', {
        		url: e.target.result
        	});
        	console.log("show it")
        	$('.upload-result-avatar').show();
        }
        reader.readAsDataURL(input.files[0]);
    }
    else {
        alert("Sorry - you're browser doesn't support the FileReader API");
    }
}
```

...and the HTML:

```
      <div id="upload-avatar" style="background:green">Upload Avatar:</div>
      <div id="avatar-upload-wrapper">
        <div id="view-avatar">
          <img id="currentAvatar" src="<%if user.avatar_image%><%avatar_image%><%else%>/images/no-profile-image.png<%endif%>" />
        </div>
        <div class="actions" id="uploader-avatar">
            <a class="btn file-btn">
                <span>Upload</span>
                <input type="file" id="upload" value="Choose a file" accept="image/*" />
            </a>
            <button class="upload-result-avatar">Save Avatar</button>
        </div>
      </div>
```

However, when I run that - I get a weird crop (totally wrong size, and even the wrong place. Here are a couple of screenshots:

![2016-04-19 at 16-16-21](https://cloud.githubusercontent.com/assets/2000191/14644491/294cb718-064a-11e6-9f49-35f8e74f3aa9.jpg)

...and another example:

![2016-04-19 at 16-19-57](https://cloud.githubusercontent.com/assets/2000191/14644622/95d4b930-064a-11e6-9d44-0e9e25b52070.jpg)

The silly thing, is that it works fine with the exact same image:

![2016-04-19 at 16-21-11](https://cloud.githubusercontent.com/assets/2000191/14644689/d28bd016-064a-11e6-9551-41f78d13191e.jpg)

...you can give it a whirl here:

http://andyapp.com.nmsrv.com/test/test.html

Any ideas? I'm drawing a blank :/

Cheers

Andy

**Comment by thedustinsmith** [comment]
When I'm testing out on your site it's working fine.  Could you possibly explain a little more what you mean when you say:

> The silly thing, is that it works fine with the exact same image


What size image are you uploading?  What browser are you using?  Any other details could help.  Again, I'm not seeing the issue you're describing.

**Comment by youradds** [comment]
Thanks for the quick reply :) 

> When I'm testing out on your site it's working fine. Could you possibly explain a little more what you mean when you say:
> 
>     The silly thing, is that it works fine with the exact same image
> 

I just mean, that it works fine (as I would expect), as this page: http://andyapp.com.nmsrv.com/test/test.html , which is the same code as I'm using within the website (I wanted to see if it had the same issue, but it doesn't)

>What size image are you uploading? What browser are you using? 

Its in FF (latest), but I think its doing it in other browsers too. I've tried with different images - and they all seem to have the same issue.

But as I said, the same code here: http://andyapp.com.nmsrv.com/test/test.html .... works just fine for me. But when put inside the main website its going on, thats when it mis-behaves

I can get you access to view it if you'd like? (the site is hidden behind an IP block, as its the dev site). Please email me on andy@ultranerds.co.uk if you are happy to do that :)

Thanks!

Andy

**Comment by thedustinsmith** [comment]
Ah - I see.  Sorry I misunderstood.

Is the styling the exact same between the link you provided and your dev site?

**Comment by youradds** [comment]
Ah interesting - the CSS from the main site does seem to be the culprit. I'm just going to duplicate it, removing bit by bit, and see when it starts working as it should - so I can try and pin down what is causing the issue

Cheers

Andy

**Comment by youradds** [comment]
Ah man, found it!!!!

`img{ max-width:100%; max-height:100% }`

Not sure why they had that in there! I'm now trying to figure out how I can get around that. Trying this:

```
.cr-boundary img {
	width: auto !important;
	height: auto !important;
}
```

...yet that doesn't seem to have any affect :(

**Comment by thedustinsmith** [comment]
You could do
```
.cr-boundary img { max-height: none; max-width: none; }
```

Usually that css is in there for mobile images.  We haven't found a work around for croppie yet.

**Comment by youradds** [comment]
A sweet, that did the trick - thanks :)

**Comment by thedustinsmith** [comment]
Glad to hear it.

**Comment by erperejildo** [comment]
I still having the same problem, the cropped image doesn't have the correct position and css doesn't help me because I'm using canvas for that.

Here is my call (a simple one):
```
$scope.edit.result({
        type: 'canvas',
        format: 'jpeg'
      }).then(function (resp) {
        $scope.addSrc(resp);
});
```

The cropped image always moves to top-left. Do I have to do anything else?

BTW, good job with the library, I'm testing it with Cordova + AngularJS and it's going well on my phone.

**Comment by thedustinsmith** [comment]
How far off are we talking?  1px?  10px?  50?

**Comment by erperejildo** [comment]
I'm not sure, it depends, sometimes is more or less, but I did two screenshots:
Before:
https://dl2.pushbulletusercontent.com/RRwaWCZ7AptfbvNmc9wkPmJd9pnb1t01/Screenshot_2016-05-23-21-07-28.png
After:
https://dl2.pushbulletusercontent.com/as0d5R2eFYGQAhiQdgjH1kQizUg3zAPc/Screenshot_2016-05-23-21-09-34.png

**Comment by erperejildo** [comment]
Another test with a landscape picture and taking 100% height.
Before:
https://dl2.pushbulletusercontent.com/boVDrzPL1yGJKsMStAwPomQVi2LgmbzQ/Screenshot_2016-05-23-21-36-00.png
After:
https://dl2.pushbulletusercontent.com/RE6KjgjqOZVxnKRq2vuSeCjf2e1f0HxQ/Screenshot_2016-05-23-21-36-47.png

Besides, I've had a weird problem this time doing the screenshots before crop it:
https://dl2.pushbulletusercontent.com/fXOg50V3oC86oo8EMVPbBRvm72Xg0pmp/Screenshot_2016-05-23-21-35-21.png

If you need different test just tell me

**Comment by thedustinsmith** [comment]
Make sure that you're initializing your croppie when your croppie element is visible.  Looks like you may have a mobile app.  Is the croppie element visible when the page is loaded?  When you click on the tab that shows the croppie, do you then call `$('.a-croppie').croppie({viewport: {}, boundary: {}, etc})`?  Or all you calling it on page load when it's not visible?

**Comment by erperejildo** [comment]
Yes, the croppie doesn't load with the page, I have to select the picture first from my gallery and them, I initialize it with the code before.

**Comment by erperejildo** [comment]
I found the problem!

I was triyng to push to my url a new test whit 2 different scenarios, so I had to create 2 croppie elements. My id always has been called "editing" so I created in this time "editing1" and "editing2" for this test.
And them it worked. Only with "editing" id I've got the problem if I use a different one works perfectly.

BTW in the html which I tried I only had the tag to instilase the croppie and a button, no more elements. 

Weird, isn't?
